cgt_calc/main.py

```diff
@@ -276,107 +391,92 @@ def convert_to_hmrc_transactions(
         print(f"Interest: £{round_decimal(interest, 2)}")
         print(f"Disposal proceeds: £{round_decimal(total_sells, 2)}")
         print("")
-        return acquisition_list, disposal_list
 
-    @staticmethod
     def process_acquisition(
-        acquisition_list: HmrcTransactionLog,
-        bed_and_breakfast_list: HmrcTransactionLog,
-        portfolio: dict[str, tuple[Decimal, Decimal]],
+        self,
         symbol: str,
         date_index: datetime.date,
     ) -> list[CalculationEntry]:
         """Process single acquisition."""
-        acquisition_quantity, acquisition_amount, acquisition_fees = astuple(
-            acquisition_list[date_index][symbol]
-        )
-        original_acquisition_amount = acquisition_amount
-        if symbol not in portfolio:
-            portfolio[symbol] = (Decimal(0), Decimal(0))
-        current_quantity, current_amount = portfolio[symbol]
+        acquisition = self.acquisition_list[date_index][symbol]
+        modified_amount = acquisition.amount
+        current_quantity, current_amount = self.final_portfolio[symbol]
         calculation_entries = []
-
         # Management fee transaction can have 0 quantity
-        assert acquisition_quantity >= 0
+        assert acquisition.quantity >= 0
         # Stock split can have 0 amount
-        assert acquisition_amount >= 0
-        bed_and_breakfast_quantity = Decimal(0)
-        bed_and_breakfast_amount = Decimal(0)
+        assert acquisition.amount >= 0
+
+        bnb_acquisition = HmrcTransactionData()
         bed_and_breakfast_fees = Decimal(0)
```

Comment:
in the original code it was recalculated as 
```
                bed_and_breakfast_fees = (
                    acquisition_fees * bed_and_breakfast_quantity / acquisition_quantity
                )
```
I do not know enough on BnB to make a judgment if it will be the same, if you confirm I am fine with that :)

Reply:
Right, let's keep it then.